tools/ncf_constraints.py

```diff
@@ -0,0 +1,107 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+if sys.version_info < (3,):
+  string_type = unicode
+else:
+  string_type = str
+
+### Constraint checking function
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
+
+def not_match_regex(parameter, regex):
+  return not match_regex(parameter, regex)
+
+def allow_empty_string(parameter, allow):
+  return allow or not_match_regex(parameter, "^$")
+
+def allow_whitespace_string(parameter, allow):
+  return allow or not_match_regex(parameter, r'^\s') or not_match_regex(parameter, r'\s$')
+
+
+constraints = {
+    "list" : {
+        "check" : from_list
+      , "type"  : list
+    }
+  , "allow_whitespace_string" : {
+        "check" : allow_whitespace_string
+      , "type"  : bool
+    }
+  , "allow_empty_string" : {
+        "check" : allow_empty_string
+      , "type"  : bool
+    }
+  # use unicode for regexs, since they will be parsed as unicode ...
+  , "regex" :  {
+        "check" : match_regex
+      , "type"  : string_type
+    }
+  , "not_regex":  {
+        "check" : not_match_regex
+      , "type"  : string_type
+    }
+  , "max_length" :  {
+        "check" : max_length
+      , "type"  : int
+    }
+  , "min_length" :  {
+        "check" : min_length
+      , "type"  : int
+    }
+}
+
+def check_parameter(parameter, parameter_constraints):
+  """Checks that a parameter is ok with the constraint passed as parameter"""
+
+  result = True
+  errors = []
+
+  for (constraint_name, constraint_value) in parameter_constraints.iteritems():
+    if not constraint_name in constraints:
+      result = False
+      errors.append (constraint_name + " is not a valid constraint")
```

Comment:
Please don't store *config* errors (from the generic method's implementation) in the same field as *validation* errors (from the user provided values). The "errors" field I created is used to have useful logging output in ncf.py (something like """Value bla doesn't match constraints " ".join("errors")""").

I propose you rename them "failed_constraints" and keep the failed constraints in that one, and use the 'errors' field as a human readable message to log/return via API.